### ISSUE TITLE
NPE when removing an item from the adapter

### ISSUE THREAD

**Issue by franciscofranco** [mikepenz/FastAdapter#58]
```javascript
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
    at com.mikepenz.fastadapter.FastAdapter.getPreItemCount(FastAdapter.java:570)
    at com.mikepenz.fastadapter.adapters.ItemAdapter.remove(ItemAdapter.java:332)
    at com.mikepenz.fastadapter.adapters.FastItemAdapter.remove(FastItemAdapter.java:225)
```

```
02-27 02:15:38.524 30132-30132/liam.franco.selene D/liam.franco.selene: adapter item count 2
02-27 02:15:38.524 30132-30132/liam.franco.selene D/liam.franco.selene: layout pos: 0
```

The code is a simple getAdapter().remove(int) as you can see the adapter had plenty items to remove position 0. And that's after I run a simple cycle to make the specific item I want to remove is inside the adapter, so there's no way it should fail.

On the latest release 1.2.0.

Thanks for your work.

**Comment by mikepenz** [comment]
@franciscofranco could you please post the code which caused this? I would really love to reproduce this on my side. 

Seems really weird, as at this line there's nothing which could be null:
https://github.com/mikepenz/FastAdapter/blob/develop/library/src/main/java/com/mikepenz/fastadapter/FastAdapter.java#L570
(The `mAdapterSizes` is a `TreeMap` which get's never set to null)

**Comment by mattiasbe** [comment]
Looks to me like the argument is null in your code @franciscofranco , not something in FastAdapter?
That line is just the first to try and use it as a primitive (auto unboxing).

**Comment by mikepenz** [comment]
@mattiasbe thanks for answering. And yes you are right. Just tried it and seems like the exact cause for this. Totally missed this. 
 
<img width="511" alt="bildschirmfoto 2016-02-27 um 15 55 09" src="https://cloud.githubusercontent.com/assets/1476232/13373555/19733d6a-dd6b-11e5-89d3-3316ad867fd4.png">

**Comment by franciscofranco** [comment]
Makes sense. I eventually changed my code and couldn't reproduce this anymore. I'm using an event bus to pass some stuff from inside the ViewHolder to another class, but I wasn't unregistering the object from the bus if the object was removed, so I was trying to remove a null object from the adapter.

Thanks & sorry for the false alarm, I just posted in case I stumbled into some weird bug that I wouldn't be able to reproduce sometime in the future :)
